src/main/java/tutorly/logic/commands/EditSessionCommand.java

```diff
@@ -0,0 +1,175 @@
+     * @param editSessionDescriptor details to edit the session with
+     */
+    public EditSessionCommand(int id, EditSessionDescriptor editSessionDescriptor) {
+        requireNonNull(editSessionDescriptor);
+
+        this.sessionId = id;
+        this.editSessionDescriptor = new EditSessionDescriptor(editSessionDescriptor);
+    }
+
+    /**
+     * Creates and returns a {@code Session} with the details of {@code sessionToEdit}
+     * edited with {@code editSessionDescriptor}. The ID of the session cannot be edited.
+     */
+    private static Session createEditedSession(Session sessionToEdit, EditSessionDescriptor editSessionDescriptor) {
+        assert sessionToEdit != null;
+
+        Timeslot updatedDate = editSessionDescriptor.getTimeslot().orElse(sessionToEdit.getTimeslot());
+        String updatedSubject = editSessionDescriptor.getSubject().orElse(sessionToEdit.getSubject());
+
+        Session newSession = new Session(updatedDate, updatedSubject);
+        newSession.setId(sessionToEdit.getId());
+
+        return newSession;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        Optional<Session> sessionToEdit = model.getSessionById(sessionId);
+        if (sessionToEdit.isEmpty()) {
+            throw new CommandException(Messages.MESSAGE_SESSION_NOT_FOUND);
+        }
+
+        Session editedSession = createEditedSession(sessionToEdit.get(), editSessionDescriptor);
+
+        if (model.hasOverlappingSession(editedSession)) {
+            throw new CommandException(Messages.MESSAGE_SESSION_OVERLAP);
+        }
+
+        model.setSession(sessionToEdit.get(), editedSession);
+        model.updateFilteredSessionList(FILTER_SHOW_ALL_SESSIONS);
+
+        return new CommandResult.Builder(String.format(MESSAGE_EDIT_SESSION_SUCCESS, Messages.format(editedSession)))
+                .withTab(Tab.SESSION)
+                .build();
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        if (!(other instanceof EditSessionCommand)) {
+            return false;
+        }
+
+        EditSessionCommand e = (EditSessionCommand) other;
+        return sessionId == e.sessionId
+                && editSessionDescriptor.equals(e.editSessionDescriptor);
+    }
+
+    /**
+     * Stores the details to edit the session with. Each non-empty field value will replace the
+     * corresponding field value of the session.
+     */
+    public static class EditSessionDescriptor {
+        private Timeslot timeslot;
+        private String subject;
+
+        public EditSessionDescriptor() {
+        }
+
+        /**
+         * Copy constructor.
+         */
+        public EditSessionDescriptor(EditSessionDescriptor toCopy) {
+            setTimeslot(toCopy.timeslot);
+            setSubject(toCopy.subject);
+        }
+
```

Comment:
Suggested convenience constructor for consistency with `EditPersonDescriptor` that also makes writing the reverse command easier
```suggestion

        /**
        * Returns a {@code EditSessionDescriptor} with the same values as {@code session}.
        */
        public static EditSessionDescriptor fromSession(Session session) {
            EditSessionDescriptor descriptor = new EditSessionDescriptor();
            descriptor.setTimeslot(session.getTimeslot());
            descriptor.setSubject(session.getSubject());
            return descriptor;
        }

```